modules/javafx.graphics/src/test/java/test/javafx/scene/CssStyleHelperTest.java

```diff
@@ -657,4 +663,99 @@ public void initialNodeWithUserSetValueShouldNotResetValuesOnOtherNodesWithoutOv
         assertEquals(new Insets(10), a.getPadding());
         assertEquals(new Insets(4), b.getPadding());
     }
+
+    private static final String GRAY_STYLESHEET = ".my-pane {-fx-background-color: #808080}";
+    private static final String GRAY_INDIRECT_STYLESHEET = ".root {-fx-base: #808080} .my-pane {-fx-background-color: -fx-base}";
+    private static final String RED_STYLESHEET = ".my-pane {-fx-background-color: red}";
+    private static final String RED_INDIRECT_STYLESHEET = ".root {-fx-base: red} .my-pane {-fx-background-color: -fx-base}";
+    private static final String FX_BASE_GREEN_STYLESHEET = ".root {-fx-base: green}";
+    private static final String FX_BASE_GRAY_STYLESHEET = ".root {-fx-base: #808080}";
+
+    /**
+     * All cases will lead to a neutral gray color #808080.
+     *
+     * UA = USER_AGENT
+     */
+    enum OverrideCases {
+        // User agent styles win when not overridden directly or indirectly:
+        UA(GRAY_STYLESHEET, null, null, null),
+        INDIRECT_UA(GRAY_INDIRECT_STYLESHEET, null, null, null),
+
+        // Property wins when not directly overridden by author or inline style:
+        PROPERTY(null, Color.web("#808080"), null, null),
+        PROPERTY_OVERRIDES_UA(RED_STYLESHEET, Color.web("#808080"), null, null),
+        PROPERTY_OVERRIDES_INDIRECT_UA(RED_INDIRECT_STYLESHEET, Color.web("#808080"), null, null),
+
+        // Property wins even if indirectly overridden by author or inline style (resolving of a lookup does not change priority of the user agent style)::
```

Comment:
You might want to remove the double colon at the end of the sentence.

Reply:
Fixed, thanks